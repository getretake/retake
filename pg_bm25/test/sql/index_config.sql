-- Invalid create_bm25
CALL paradedb.create_bm25(
	index_name => 'index_config',
	table_name => 'index_config'
);
CALL paradedb.create_bm25(
	index_name => 'index_config',
	table_name => 'index_config',
	key_field => 'id'
);
CALL paradedb.create_bm25(
	index_name => 'index_config',
	table_name => 'index_config',
	key_field => 'id',
	invalid_field => '{}'		
);

-- Default text field
CALL paradedb.create_bm25(
	index_name => 'index_config',
	table_name => 'index_config',
	schema_name => 'paradedb',
	key_field => 'id',
	text_fields => '{"description": {}}'
);
SELECT * from paradedb.schema_bm25('index_config_bm25_index');
CALL paradedb.drop_bm25('index_config');

-- Text field with options
CALL paradedb.create_bm25(
	index_name => 'index_config',
	table_name => 'index_config',
	schema_name => 'paradedb',
	key_field => 'id',
	text_fields => '{"description": {"fast": true, "tokenizer": { "type": "en_stem" }, "record": "freq", "normalizer": "raw"}}'
);
SELECT * from paradedb.schema_bm25('index_config_bm25_index');
CALL paradedb.drop_bm25('index_config');

-- Multiple text fields
CALL paradedb.create_bm25(
	index_name => 'index_config',
	table_name => 'index_config',
	schema_name => 'paradedb',
	key_field => 'id',
	text_fields => '{"description": {fast: true, tokenizer: { type: "en_stem" }, record: "freq", normalizer: "raw"}, category: {}}'
);
SELECT * from paradedb.schema_bm25('index_config_bm25_index');
CALL paradedb.drop_bm25('index_config');

-- Default numeric field
CALL paradedb.create_bm25(
	index_name => 'index_config',
	table_name => 'index_config',
	schema_name => 'paradedb',
	key_field => 'id',
	numeric_fields => '{"rating": {}}'
);
SELECT * from paradedb.schema_bm25('index_config_bm25_index');
CALL paradedb.drop_bm25('index_config');

-- Numeric field with options
CALL paradedb.create_bm25(
	index_name => 'index_config',
	table_name => 'index_config',
	schema_name => 'paradedb',
	key_field => 'id',
	numeric_fields => '{"rating": {"fast": false}}'
);
SELECT * from paradedb.schema_bm25('index_config_bm25_index');
CALL paradedb.drop_bm25('index_config');

-- Default boolean field
CALL paradedb.create_bm25(
	index_name => 'index_config',
	table_name => 'index_config',
	schema_name => 'paradedb',
	key_field => 'id',
	boolean_fields => '{"in_stock": {}}'
);
SELECT * from paradedb.schema_bm25('index_config_bm25_index');
CALL paradedb.drop_bm25('index_config');

-- Boolean field with options
CALL paradedb.create_bm25(
	index_name => 'index_config',
	table_name => 'index_config',
	schema_name => 'paradedb',
	key_field => 'id',
	boolean_fields => '{"in_stock": {"fast": false}}'
);
SELECT * from paradedb.schema_bm25('index_config_bm25_index');
CALL paradedb.drop_bm25('index_config');

-- Default Json field
CALL paradedb.create_bm25(
	index_name => 'index_config',
	table_name => 'index_config',
	schema_name => 'paradedb',
	key_field => 'id',
	json_fields => '{"metadata": {}}'
);
SELECT * from paradedb.schema_bm25('index_config_bm25_index');
CALL paradedb.drop_bm25('index_config');

-- Json field with options
CALL paradedb.create_bm25(
	index_name => 'index_config',
	table_name => 'index_config',
	schema_name => 'paradedb',
	key_field => 'id',
	json_fields => '{metadata: {fast: true, expand_dots: false, tokenizer: { type: "raw" }, normalizer: "raw"}}'
);
SELECT * from paradedb.schema_bm25('index_config_bm25_index');
CALL paradedb.drop_bm25('index_config');

-- Multiple fields
CALL paradedb.create_bm25(
	index_name => 'index_config',
	table_name => 'index_config',
	schema_name => 'paradedb',
	key_field => 'id',
	text_fields => '{description: {}, category: {}}',
	numeric_fields => '{rating: {}}',
	boolean_fields => '{in_stock: {}}',
	json_fields => '{metadata: {}}'
);
SELECT * from paradedb.schema_bm25('index_config_bm25_index');
CALL paradedb.drop_bm25('index_config');
