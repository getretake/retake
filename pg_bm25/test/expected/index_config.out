-- Invalid CREATE INDEX
CREATE INDEX idxindexconfig ON index_config USING bm25 ((index_config.*));
ERROR:  no text_fields, numeric_fields, boolean_fields, or json_fields were specified
CREATE INDEX idxindexconfig ON index_config USING bm25 ((index_config.*)) WITH (invalid_field='{}');
ERROR:  unrecognized parameter "invalid_field"
-- Default text field
CREATE INDEX idxindexconfig ON index_config USING bm25 ((index_config.*)) WITH (text_fields='{"description": {}}');
SELECT * from paradedb.schema_bm25('idxindexconfig');
    name     | field_type | stored | indexed | fast | fieldnorms | expand_dots | tokenizer | record | normalizer
-------------+------------+--------+---------+------+------------+-------------+-----------+--------+------------
 description | Str        | t      | t       | f    | t          |             | default   | basic  |
 heap_tid    | U64        | t      | t       | f    | t          |             |           |        |
(2 rows)

DROP INDEX idxindexconfig;
-- Text field with options
CREATE INDEX idxindexconfig ON index_config USING bm25 ((index_config.*)) WITH (text_fields='{"description": {"fast": true, "tokenizer": "en_stem", "record": "freq", "normalizer": "raw"}}');
SELECT * from paradedb.schema_bm25('idxindexconfig');
    name     | field_type | stored | indexed | fast | fieldnorms | expand_dots | tokenizer | record | normalizer
-------------+------------+--------+---------+------+------------+-------------+-----------+--------+------------
 description | Str        | t      | t       | t    | t          |             | en_stem   | freq   | raw
 heap_tid    | U64        | t      | t       | f    | t          |             |           |        |
(2 rows)

DROP INDEX idxindexconfig;
-- Multiple text fields
CREATE INDEX idxindexconfig ON index_config USING bm25 ((index_config.*)) WITH (text_fields='{"description": {"fast": true, "tokenizer": "en_stem", "record": "freq", "normalizer": "raw"}, "category": {}}');
SELECT * from paradedb.schema_bm25('idxindexconfig');
    name     | field_type | stored | indexed | fast | fieldnorms | expand_dots | tokenizer | record | normalizer
-------------+------------+--------+---------+------+------------+-------------+-----------+--------+------------
 description | Str        | t      | t       | t    | t          |             | en_stem   | freq   | raw
 category    | Str        | t      | t       | f    | t          |             | default   | basic  |
 heap_tid    | U64        | t      | t       | f    | t          |             |           |        |
(3 rows)

DROP INDEX idxindexconfig;
-- Default numeric field
CREATE INDEX idxindexconfig ON index_config USING bm25 ((index_config.*)) WITH (numeric_fields='{"rating": {}}');
SELECT * from paradedb.schema_bm25('idxindexconfig');
   name   | field_type | stored | indexed | fast | fieldnorms | expand_dots | tokenizer | record | normalizer
----------+------------+--------+---------+------+------------+-------------+-----------+--------+------------
 rating   | I64        | t      | t       | t    | f          |             |           |        |
 heap_tid | U64        | t      | t       | f    | t          |             |           |        |
(2 rows)

DROP INDEX idxindexconfig;
-- Numeric field with options
CREATE INDEX idxindexconfig ON index_config USING bm25 ((index_config.*)) WITH (numeric_fields='{"rating": {"fast": false}}');
SELECT * from paradedb.schema_bm25('idxindexconfig');
   name   | field_type | stored | indexed | fast | fieldnorms | expand_dots | tokenizer | record | normalizer
----------+------------+--------+---------+------+------------+-------------+-----------+--------+------------
 rating   | I64        | t      | t       | f    | f          |             |           |        |
 heap_tid | U64        | t      | t       | f    | t          |             |           |        |
(2 rows)

DROP INDEX idxindexconfig;
-- Default boolean field
CREATE INDEX idxindexconfig ON index_config USING bm25 ((index_config.*)) WITH (boolean_fields='{"in_stock": {}}');
SELECT * from paradedb.schema_bm25('idxindexconfig');
   name   | field_type | stored | indexed | fast | fieldnorms | expand_dots | tokenizer | record | normalizer
----------+------------+--------+---------+------+------------+-------------+-----------+--------+------------
 in_stock | Bool       | t      | t       | t    | f          |             |           |        |
 heap_tid | U64        | t      | t       | f    | t          |             |           |        |
(2 rows)

DROP INDEX idxindexconfig;
-- Boolean field with options
CREATE INDEX idxindexconfig ON index_config USING bm25 ((index_config.*)) WITH (boolean_fields='{"in_stock": {"fast": false}}');
SELECT * from paradedb.schema_bm25('idxindexconfig');
   name   | field_type | stored | indexed | fast | fieldnorms | expand_dots | tokenizer | record | normalizer
----------+------------+--------+---------+------+------------+-------------+-----------+--------+------------
 in_stock | Bool       | t      | t       | f    | f          |             |           |        |
 heap_tid | U64        | t      | t       | f    | t          |             |           |        |
(2 rows)

DROP INDEX idxindexconfig;
-- Default Json field
CREATE INDEX idxindexconfig ON index_config USING bm25 ((index_config.*)) WITH (json_fields='{"metadata": {}}');
SELECT * from paradedb.schema_bm25('idxindexconfig');
   name   | field_type | stored | indexed | fast | fieldnorms | expand_dots | tokenizer | record | normalizer
----------+------------+--------+---------+------+------------+-------------+-----------+--------+------------
 metadata | JsonObject | t      | t       | f    | f          | t           | default   | basic  |
 heap_tid | U64        | t      | t       | f    | t          |             |           |        |
(2 rows)

DROP INDEX idxindexconfig;
-- Json field with options
CREATE INDEX idxindexconfig ON index_config USING bm25 ((index_config.*)) WITH (json_fields='{"metadata": {"fast": true, "expand_dots": false, "tokenizer": "raw", "normalizer": "raw"}}');
SELECT * from paradedb.schema_bm25('idxindexconfig');
   name   | field_type | stored | indexed | fast | fieldnorms | expand_dots | tokenizer | record | normalizer
----------+------------+--------+---------+------+------------+-------------+-----------+--------+------------
 metadata | JsonObject | t      | t       | t    | f          | f           | raw       | basic  | raw
 heap_tid | U64        | t      | t       | f    | t          |             |           |        |
(2 rows)

DROP INDEX idxindexconfig;
-- Multiple fields
CREATE INDEX idxindexconfig ON index_config USING bm25 ((index_config.*)) WITH (text_fields='{"description": {}, "category": {}}', numeric_fields='{"rating": {}}', boolean_fields='{"in_stock": {}}', json_fields='{"metadata": {}}');
SELECT * from paradedb.schema_bm25('idxindexconfig');
    name     | field_type | stored | indexed | fast | fieldnorms | expand_dots | tokenizer | record | normalizer
-------------+------------+--------+---------+------+------------+-------------+-----------+--------+------------
 description | Str        | t      | t       | f    | t          |             | default   | basic  |
 rating      | I64        | t      | t       | t    | f          |             |           |        |
 category    | Str        | t      | t       | f    | t          |             | default   | basic  |
 in_stock    | Bool       | t      | t       | t    | f          |             |           |        |
 metadata    | JsonObject | t      | t       | f    | f          | t           | default   | basic  |
 heap_tid    | U64        | t      | t       | f    | t          |             |           |        |
(6 rows)

DROP INDEX idxindexconfig;
